### PR TITLE
codeintel: Remove dead code (api package, dbstore, lsifstore)

### PR DESCRIPTION
**Pull request by efritz**

### CODE REVIEW

File: doc/admin/observability/alert_solutions.md [hunk 1]
```diff
@@ -536,50 +536,6 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
-```json
-"observability.silenceAlerts": [
-  "warning_frontend_codeintel_api_errors"
-]
-```
-
-<sub>*Managed by the [Sourcegraph Code-intelligence team](https://about.sourcegraph.com/handbook/engineering/code-intelligence).*</sub>
-
-<br />
-
```
Comment (Strum355) [inline doc/admin/observability/alert_solutions.md:582]: If these are removed, are there more granular metrics that we track instead?

Comment (efritz) [reply doc/admin/observability/alert_solutions.md:582]: Yes, each time we invoke a method on any external system as well as each resolver itself

Comment (efritz) [reply doc/admin/observability/alert_solutions.md:582]: (These were basically duplicates of the resolver metrics in the first place)

**Review by Strum355 — APPROVED** [review]
:hocho:

**Comment by daxmc99** [comment]
There appears to be some code added here?

**Comment by efritz** [comment]
@daxmc99 My target branch changed and messed some stuff up, trying to work through it.

**Pull request merged**